src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -100,6 +100,8 @@
 
     /**
      * Additional node settings loaded by the plugin
+     *
+     * @param settings The already configured node settings
      */
-    Settings additionalSettings();
+    Settings additionalSettings(Settings settings);
```

Comment:
This will break all plugins and will force all authors to update their plugins.
Not sure if there are any other option but if not we should mark it as a breaking change.

Can we add it only to `AbstractPlugin`? I mean add it as a new function but not replacing the existing signature.